src/Controller/CategoriesController.php

```diff
@@ -51,12 +52,37 @@ public static function create(ContainerInterface $container) {
    *   Json Array with uuid's.
    */
   public function list(string $type) {
+    $user = $this->currentUser();
+    $pattern = 'virtual_y';
+    // Get list of vy roles for current user.
+    $y_roles = array_filter($user->getRoles(), function ($role) use ($pattern) {
+      return strpos($role, $pattern) !== FALSE;
+    });
+
     $query = $this->database->select('node__field_gc_video_category', 'n');
+    $query->leftJoin('node_field_data', 'nd', 'n.entity_id = nd.nid');
     $query->leftJoin('taxonomy_term_data', 't', 't.tid = n.field_gc_video_category_target_id');
     $query->leftJoin('taxonomy_term_field_data', 'tf', 't.tid = tf.tid');
     $query->condition('n.bundle', $type);
     $query->condition('t.vid', 'gc_category');
     $query->condition('tf.status', 1);
+
+    if (!empty($y_roles)) {
+      $or_group = $query->orConditionGroup();
+      foreach ($y_roles as $role) {
+        if ($role == 'virtual_y') {
+          // The rest of roles already contains `virtual_y` string, so we need
+          // more strict condition here.
+          $or_group->condition('nd.field_vy_permission', $role, 'LIKE');
```

Comment:
Any reason to use 'LIKE' but not '='?

Reply:
DB value can contain multiple roles separated by a comma, for example: `virtual_y,virtual_y_premium`

Reply:
But for the case above you have no `'%'` in the value which will be compared with the DB value
So tactically it's equal you're using the `'='` operator

Reply:
in this case yes, we can use `=` instead

Reply:
Let's do this so -- for the sake of a clarity

Reply:
Oh, btw, seems it's better to use the same comparison conditions for other roles either:
what if there would be `virtual_y_premium` and `virtual_y_premium_extra` roles?

Reply:
Fixed